### ISSUE TITLE
[Bug] cannot resolve symbol 'build_stl_str_hl'

### ISSUE THREAD

**Issue by qgymib** [famiu/feline.nvim#143]
**Neovim version**
```
NVIM v0.5.1
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: C:/Program Files (x86)/Microsoft Visual Studio/2017/Enterprise/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe /DWIN32 /D_WINDOWS /W3 -DNVIM_TS_HAS_SET_MATCH_LIMIT /MD /Zi /O2 /Ob1 /DNDEBUG /W3 -D_CRT_SECURE_NO_WARNINGS -D_CRT_NONSTDC_NO_DEPRECATE -DWIN32 -D_WIN32_WINNT=0x0600 -DINCLUDE_GENERATED_DECLARATIONS -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -ID:/a/neovim/neovim/build/config -ID:/a/neovim/neovim/src -ID:/a/neovim/neovim/nvim-deps/usr/include -ID:/a/neovim/neovim/build/src/nvim/auto -ID:/a/neovim/neovim/build/include
Compiled by runneradmin@fv-az152-633

Features: -acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM\sysinit.vim"
  fall-back for $VIM: "C:/Program Files/nvim/share/nvim"

Run :checkhealth for more info
```

**Feline version**
master branch with newest commit: `227566a346a975f8f301fc21a0d02d4068f78b0a`.
But it seems tag v0.3 also have this issue. Other version is not tested.

**Describe the bug**
startup failed with following error message:
```
E5108: Error executing lua ...k\packer\start\feline.nvim\lua\feline\statusline_ffi.lua:38: cannot resolve symbol 'build_stl_str_hl': <d5>Ҳ<bb><b5><bd>ָ<b6><a8><b5>ĳ<cc><d0><f2><a1><a3>
```

**To Reproduce**
Use following init.lua, run `:PackerInstall` and `:PackerCompile`, then restart nvim and the error occur:
```
-- ensure pakcer is installed
local function ensure_packer_installed()
    -- check if packer is installed as startup plugin
    local install_path = vim.fn.stdpath('data')..'/site/pack/packer/start/packer.nvim'
    if vim.fn.empty(vim.fn.glob(install_path)) <= 0 then
        return
    end
    -- if packer is also not installed as opt, install it
    local install_opt_path = vim.fn.stdpath('data')..'/site/pack/packer/opt/packer.nvim'
    if vim.fn.empty(vim.fn.glob(install_opt_path)) > 0 then
        vim.fn.system({'git', 'clone', '--depth', '1', 'https://github.com/wbthomason/packer.nvim', install_opt_path})
    end
    -- load packer.nvim
    vim.cmd 'packadd packer.nvim'
end
ensure_packer_installed()

require('packer').startup({function(use)
    use 'wbthomason/packer.nvim'

    use {
        'famiu/feline.nvim',
        requires = 'kyazdani42/nvim-web-devicons',
        config = function()
            require('feline').setup()
        end
    }
end
})
```

**Expected behavior**
No error should occur.

**Does this error occur in the minimal init file?**
Not tested. the minimal init file is not for windows.

**Provide modified minimal_init.lua**
Just like above.

**Screenshots**
![image](https://user-images.githubusercontent.com/4520263/135952293-c4ea92ea-7df3-4b20-83d8-cbed8490d5b8.png)

**Additional context**
none.

**Comment by famiu** [comment]
If I were to guess, the issue is probably due to the fact that you use Windows. And if that's the case, I don't think I can help with it since I don't use Windows and thus have no way to check what fixes it. If you can make a PR that fixes your issue, that'd be welcome

**Comment by qgymib** [comment]
> @qgymib @UnderCooled does #150 fix your issues?

Not yet, still got the same error. Tested with develop branch with newest commit: `6b49a17418de65a6fedbcdea6ed78b51432d2719`

```
E5108: Error executing lua ...k\packer\start\feline.nvim\lua\feline\statusline_ffi.lua:38: cannot resolve symbol 'build_stl_str_hl': <d5>Ҳ<bb><b5><bd>ָ<b6><a8><b5>ĳ<cc><d0><f2><a1><a3>
```

**Comment by qgymib** [comment]
I'm sorry, I test with wrong branch, thought the fix was merge into develop branch.

Now test with #150, `vim.loop.os_uname().sysname == 'Windows_NT'` fix the problem.  Compare with `Windows` is not working.

**Comment by famiu** [comment]
> I'm sorry, I test with wrong branch, thought the fix was merge into develop branch.
> 
> Now test with #150, `vim.loop.os_uname().sysname == 'Windows_NT'` fix the problem. Compare with `Windows` is not working.

Is it fixed now?

**Comment by qgymib** [comment]
Yes it is, with your commit `fix Windows check` (`b3be24e60cf0230699893d0a59cceda14a8959f8`)
